chat-widget/src/components/livechatwidget/common/endChat.ts

```diff
@@ -55,8 +55,13 @@ const prepareEndChat = async (props: ILiveChatWidgetProps, chatSDK: any, setAdap
     if (isPostChatEnabled === "true" && conversationDetails?.canRenderPostChat === Constants.truePascal) {
         const skipEndChatSDK = false;
         const skipCloseChat = true;
+        const chatSession = await chatSDK?.getCurrentLiveChatContext();
         await endChat(props, chatSDK, setAdapter, setWebChatStyles, dispatch, adapter, skipEndChatSDK, skipCloseChat, false);
-
+        if (chatSession) {
+            (chatSDK as any).chatToken = chatSession.chatToken ?? {};
+            chatSDK.requestId = chatSession.requestId;
```

Comment:
Don't we need to handle and clear the context and token after chat has ended or else it might cause issues with new chat instances.
In this case I think what we can do is clear the context and token in the close chat part

Reply:
The clearing of the context and token is already handled in the endChat method within endChat.ts file. See https://github.com/microsoft/omnichannel-chat-widget/blob/f0b26a69d61a7c3be5d023d83ad7ae7e92b06b6a/chat-widget/src/components/livechatwidget/common/endChat.ts#L101

Reply:
Since we're reusing the requestId & chatId of the conversation that was closed to download/email the transcript, I'd recommend to reset those values after the chat widget is fully closed (when download/email transcript won't be available anymore).

On `ChatSDK.endChat()`, it basically recreates a new `requestId` via `uuidv4()` & sets the `chatToken` to be `{}`. You can use these values on reset.

Reply:
I agree we should be resetting the values ideally to avoid any side effect.
Please verify if the values are being reset after fully closing chat in this case

Reply:
We are already resetting the values in the endChat method of endChat.ts:
![image](https://user-images.githubusercontent.com/13504205/201183362-2c005dfe-b0ea-4173-8611-8ae2a8e53c52.png)
endChat flow will get called when C2 officially ends the chat (chat widget minimized and download/email transcript no longer available).

I verified that the values are in fact already getting reset through console logs added. And starting new chats do not use the previous chatToken/requestId.
<img width="558" alt="image" src="https://user-images.githubusercontent.com/13504205/201183602-2e91f480-d505-44f8-a00d-05a056361972.png">